### PR TITLE
Fix cppcheck 1.87 warnings in Nexus

### PR DESCRIPTION
**Pull request by soininen**
This PR fixes `cppcheck` warnings in the Nexus module.

**To test:**

Code review.

Brings #22912 closer to closing.

*This does not require release notes* because this is an internal change.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

File: Framework/Nexus/src/NexusFileIO.cpp [hunk 1]
```diff
@@ -1277,8 +1269,7 @@ int getNexusEntryTypes(const std::string &fileName,
   // for each entry found, look for "analysis" or "definition" text data fields
   // and return value plus entry name
   for (auto &entry : entryList) {
-    //
-    stat = NXopengroup(fileH, entry.c_str(), "NXentry");
+    NXopengroup(fileH, entry.c_str(), "NXentry");
```
Comment (TTitcombe) [inline Framework/Nexus/src/NexusFileIO.cpp]: why is the `stat = ` not necessary here?

Comment (soininen) [reply Framework/Nexus/src/NexusFileIO.cpp]: The value is never used. `stat` gets reassigned immediately in the proceeding `while` loop.

Comment (TTitcombe) [reply Framework/Nexus/src/NexusFileIO.cpp]: oh yeah, I see that now :man_facepalming:

**Review by TTitcombe — DISMISSED** [review]
Code changes look good

**Review by DanNixon — CHANGES_REQUESTED** [review]

File: Framework/Nexus/src/NexusFileIO.cpp [hunk 2]
```diff
@@ -1277,8 +1269,7 @@ int getNexusEntryTypes(const std::string &fileName,
...
     // loop through field names in this entry
     while ((stat = NXgetnextentry(fileH, nxname, nxclass, &nxdatatype)) ==
```
[the remark this answers is not in the document]
Comment (soininen) [reply Framework/Nexus/src/NexusFileIO.cpp]: Nice catch! I don't think it is and it is missed by `cppcheck` as well.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**